Source/Particles/Resampling/VelocityCoincidenceThinning.H

```diff
@@ -106,10 +113,58 @@ public:
         }
     };
 
+    /**
+     * \brief Struct used to assign velocity space bin numbers to a given set
+     * of particles.
+    */
+    struct VelocityBinCalculator {
+
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        void operator() (const amrex::ParticleReal ux[], const amrex::ParticleReal uy[],
+                         const amrex::ParticleReal uz[], const unsigned int indices[],
+                         int bin_array[], int index_array[],
+                         const int cell_start, const int cell_stop) const
+        {
+            for (int i = cell_start; i < cell_stop; ++i)
+            {
+                int ii = -1, jj = -1, kk = -1;
+                if (velocity_grid_type == VelocityGridType::Spherical) {
```

Comment:
Could this check be put outside the loop so that it isn't done for every particle? Could what is being done in PR #4888 be done here?

Reply:
Sounds good! I'll update accordingly. I was not aware of that technique for reducing register pressure. Thanks for pointing it out!

Reply:
Digging a little deeper I'm not sure whether the `CompileTimeOptions` will help in this case since the velocity binning is done inside a GPU loop already. But I have moved things around some so that now we only check the grid type once per tile rather than for every particle.